ros_tcp/ros_communication_client/ros_communications.py

```diff
@@ -161,45 +168,52 @@ def __init__(
         self.verbose = verbose
 
     def __del__(self):
-        print('Destroying Subscriber') if self.verbose else None
+        print("Destroying Subscriber") if self.verbose else None
         self.unsubscribe()
         del self.tcp_connection
 
     def recieve(self):
-        while not self.stop_event.is_set():
-            rosmsg = self.tcp_connection.recv_bson()
-            self.fps_counter = (
-                FPSCounter() if self.fps_counter is None else self.fps_counter
-            )
-            # print(f"Data recieved {rosmsg['msg'].keys()}")
-            if rosmsg["op"] == "publish" and rosmsg["topic"] == self.topic_name:
-                # print("ros msg:", rosmsg["msg"]["encoding"], self.topic_name)
-                try:
-                    data = self.from_msg_type(rosmsg)
-                except Exception as e:
-                    print("ros msg:", rosmsg["msg"]["encoding"], self.topic_name)
-                    raise e
-
-                self.fps_counter.update(verbose=self.verbose)
-                if self.callback_fn is not None:
-                    self.callback_fn(data)
-                else:
-                    self.data = data
+
+        try:
```

Comment:
why does sometimes this fail?

Reply:
There was some bug in the ros_tcp code which @tusharsangam  helped to debug and fix.
During this debug we added this try-except block so we can see full traceback.

It shouldn't happen now but seeing full-traceback helps regardless because things were getting masked into try-catch else where.